sdk/tables/Azure.Data.Tables/readme.md

```diff
@@ -1,34 +1,207 @@
+
+```C# Snippet:TablesSample3QueryTables
+// Use the <see cref="TableServiceClient"> to query the service. Passing in OData filter strings is optional.
+
+Pageable<TableItem> queryTableResults = serviceClient.GetTables(filter: $"TableName eq '{tableName}'");
+
+Console.WriteLine("The following are the names of the tables in the query results:");
+
+// Iterate the <see cref="Pageable"> in order to access queried tables.
+
+foreach (TableItem table in queryTableResults)
+{
+    Console.WriteLine(table.TableName);
+}
+```
+
+Individual tables can bel deleted from the service.
+
+```C# Snippet:TablesSample1DeleteTable
+// Deletes the table made previously.
+
+serviceClient.DeleteTable(tableName);
+```
+
+### Add, Query, and Delete table entities
+
+To interact with table entities, we must first construct a `TableClient`.
+
+```C# Snippet:TablesSample2CreateTableWithTableClient
+// Construct a new <see cref="TableClient" /> using a <see cref="TableSharedKeyCredential" />.
+
+var tableClient = new TableClient(
+    new Uri(storageUri),
+    tableName,
+    new TableSharedKeyCredential(accountName, storageAccountKey));
+
+// Create the table in the service.
+tableClient.Create();
+```
+
+Let's define a new `TableEntity` so that we can add it to the table.
+
+```C# Snippet:TablesSample2CreateDictionaryEntity
+// Make a dictionary entity by defining a <see cref="TableEntity">.
+
+var entity = new TableEntity(partitionKey, rowKey)
+{
+    { "Product", "Marker Set" },
+    { "Price", 5.00 },
+    { "Quantity", 21 }
+};
+
+Console.WriteLine($"{entity.RowKey}: {entity["Product"]} costs ${entity.GetDouble("Price")}.");
+```
+
+Using the `TableClient` we can now add our new entity to the table.
+
+```C# Snippet:TablesSample2AddEntity
+// Add the newly created entity.
+
+tableClient.AddEntity(entity);
+```
+
+To inspect the set of existing table entities, we can query the table using an OData filter.
+
+```C# Snippet:TablesSample4QueryEntitiesFilter
+Pageable<TableEntity> queryResultsFilter = tableClient.Query<TableEntity>(filter: $"PartitionKey eq '{partitionKey}'");
+
+// Iterate the <see cref="Pageable"> to access all queried entities.
+
+foreach (TableEntity qEntity in queryResultsFilter)
+{
+    Console.WriteLine($"{qEntity.GetString("Product")}: {qEntity.GetDouble("Price")}");
+}
+
+Console.WriteLine($"The query returned {queryResultsFilter.Count()} entities.");
+```
+
+If we no longer need our new table entity, it can be deleted.
+
+```C# Snippet:TablesSample2DeleteEntity
+// Delete the entity given the partition and row key.
+
+tableClient.DeleteEntity(partitionKey, rowKey);
+```
 
 ## Troubleshooting
 
-Content forthcoming
+When you interact with the Azure table library using the .NET SDK, errors returned by the service correspond to the same HTTP 
+status codes returned for [REST API][keyvault_rest] requests.
+
+For example, if you try to retrieve a create a table that already exists, a `409` error is returned, indicating "Conflict".
```

Comment:
"if you try to retrieve a create a table that already exists" -> "if you try to create a table that already exists,"